### DISCUSSION TITLE
Homing switches

### DISCUSSION THREAD

**Discussion by Festivejelly** [kachurovskiy/nanoels#155] [Ideas]
Would homing switches be possible to use on the H4? I know it would need some code changes but it would be handy to home to get absolute position values. Then you could configure offsets from these.

For example homing X then doing a light cut and measuring the diameter we would be able to work out what the offset of the limit switch on X is. Similarly by homing on Z then taking a facing pass we could see what the offset would be from the Z limit switch.

Then these offsets could go in the config, and when we home the lathe our position should always be accurate.

Could use the switches that are used on CNC machines, I think they are inductive maybe?

I thought maybe backlash could cause issues but if its measured we could always compensate. Might also make it a bit easier to set the soft limits as well.

**Comment by kachurovskiy** [comment]
Hi Nicholas, yes, I thought about this as well. Since most of the time we use chucks and stock of unknown length and stickout, homing Z doesn't appear useful to me. Much easier to take a light pass and set a `0` there? Or just move the stock up to the tool while clamping.

With X, it would be really nice to be homed at the centerline - but that would require for the limit switch to be located on the centerline to account for the tool. It just seems too complex to be worth it since there's an easy way to home by taking a light pass and measuring. This also accounts for toolpost and stock flexing which won't be considered by a limit switch.

Then there's the added risk of homing procedure crashing the machine if the switch doesn't trigger as intended e.g. due to debris or misconfiguration.

**Reply by Festivejelly** [reply]
I see your point about stock length but in terms of centre line if you home on x then you make a light cut , measure and use the radius that would give you the centre line.

so lets say you home x then move near the middle to take a light cut. As we take the light cut if we observer the distance from the end stop as being 120mm and we measure the radius of the cut part to be 10mm we can be sure the centre line is always going to be at 110mm from the endstop, assuming that is you get good repeatability. Probably would require ball screws.
